File: apps/flight-planner/tests/integration_test.py
# Integration tests for flight-planner

import unittest
from unittest.mock import patch
import requests
import websocket


class TestFlightPlannerIntegration(unittest.TestCase):

    def setUp(self):
        # Setup code, if needed
        pass

    def tearDown(self):
        # Teardown code, if needed
        pass

    def test_telemetry_streaming(self):
        # Mock WebSocket connection and test telemetry streaming
        ws = websocket.WebSocket()
        ws.connect("ws://localhost:8000/telemetry")
        message = ws.recv()
        self.assertIsNotNone(message)
        ws.close()

    def test_flight_plan_api(self):
        # Test flight plan API endpoint
        response = requests.get("http://localhost:8000/api/flight-plan")
        self.assertEqual(response.status_code, 200)
        data = response.json()
        self.assertIn('flight_plan', data)

    def test_nav_data_query(self):
        # Test navigation data query API endpoint
        response = requests.get("http://localhost:8000/api/nav-data")
        self.assertEqual(response.status_code, 200)
        data = response.json()
        self.assertIn('nav_data', data)

if __name__ == '__main__':
    unittest.main()

File: apps/flight-planner/tests/integration_test.go
// Integration tests for Flight Planner

package tests

import (
	"testing"
	"github.com/stretchr/testify/assert"
	"net/http"
	"net/http/httptest"
	"encoding/json"
	"flight-planner/app"
)

func TestCreateFlightPlan(t *testing.T) {
	// Setup
	server := httptest.NewServer(app.SetupRouter())
	defer server.Close()

	// Test data
	flightPlan := map[string]interface{}{
		"origin": "JFK",
		"destination": "LAX",
		"waypoints": []string{"ORD", "DEN"},
	}

	// Convert to JSON
	jsonData, err := json.Marshal(flightPlan)
	assert.NoError(t, err)

	// Make request
	resp, err := http.Post(server.URL+"/api/flightplans", "application/json", bytes.NewBuffer(jsonData))
	assert.NoError(t, err)
	assert.Equal(t, http.StatusCreated, resp.StatusCode)

	// Cleanup
	resp.Body.Close()
}

func TestReadFlightPlan(t *testing.T) {
	// Setup
	server := httptest.NewServer(app.SetupRouter())
	defer server.Close()

	// Test data
	flightPlanID := "12345"

	// Make request
	resp, err := http.Get(server.URL + "/api/flightplans/" + flightPlanID)
	assert.NoError(t, err)
	assert.Equal(t, http.StatusOK, resp.StatusCode)

	// Parse response
	var result map[string]interface{}
	err = json.NewDecoder(resp.Body).Decode(&result)
	assert.NoError(t, err)
	assert.Equal(t, flightPlanID, result["id"])

	// Cleanup
	resp.Body.Close()
}

// Additional tests for update, delete, and other scenarios can be added similarly.
